OCP-4.X/roles/controller-kickstart/tasks/main.yml

```diff
@@ -9,21 +9,21 @@
 - name: set workdir when KUBECONFIG_AUTH_DIR_PATH is defined
   set_fact:
     workdir: "{{ KUBECONFIG_AUTH_DIR_PATH }}"
-  when: 
+  when:
     - KUBECONFIG_AUTH_DIR_PATH is defined
     - KUBECONFIG_AUTH_DIR_PATH != ""
-    
+
 - name: ensure that .kube dir exists on controller
   file:
     path: "{{ item }}"
     state: directory
   with_items:
     - /home/core/.kube
     - /root/.kube
- 
+
 - name: copy the kubeconfig to controller node
   copy:
-    src: "{{ workdir }}/auth/kubeconfig"
+    src: "{{playbook_dir}}/kubeconfig"
```

Comment:
In the case where KUBECONFIG_AUTH_DIR_PATH inventory variable is defined and has a value, is the "workdir" variable being used in this controller-kickstart role task file ?  Where do we define the "playbook_dir" variable ?  Is it in the inventory file ?

Reply:
The `playbook_dir` variable is a [magic variable](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#accessing-information-about-other-hosts-with-magic-variables) by Ansible.  In the above sense it provides the full path to a place to store the kubeconfig.

Example:
```
[akrzos@bithead scale-ci-ansible]$ ansible-playbook -vv -i inventory OCP-4.X/install.yml
```

`playbook_dir` for the above `ansible-playbook` run would result in the full path to directory `scale-ci-ansible/OCP-4.X` (Wherever scale-ci-ansible is cloned to)

Reply:
Flexy creates an archive zip file containing the kubeconfig of the cluster it just installed.  We have a job to download that zip archive and expand it on jump host.  In the inventory file, we specify the full path to the auth dir containing the kubeconfig file in variable "KUBECONFIG_AUTH_DIR_PATH", in the expanded zip archive. 

Proposed change:

- name: copy the kubeconfig to controller node
  copy:
    src: "{{playbook_dir}}/kubeconfig"
    dest: "{{ item }}"
  with_items:
    - /home/core/.kube/config
    - /root/.kube/config
  when:
    - KUBECONFIG_AUTH_DIR_PATH is undefined

# handle case where kubeconfig is in another specified dir
- name: copy the kubeconfig to controller node from specific directory
  copy:
    src: "{{workdir}}/auth/kubeconfig"
    dest: "{{ item }}"
  with_items:
    - /home/core/.kube/config
    - /root/.kube/config
  when:
    - KUBECONFIG_AUTH_DIR_PATH is defined
    - KUBECONFIG_AUTH_DIR_PATH != ""

Reply:
I believe I have addressed this.  I'll try out the install again to make sure this works in the cases mentioned.  Please re-review and let me know if this won't work for flexy.